rubocop/cops/required_inverse_of_relations.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module CustomCops
+  class RequiredInverseOfRelations < RuboCop::Cop::Base
+    MSG = 'Use inverse_of option when defining associations to prevent avoidable SQL queries and keep models in sync.'
+
+    # Optimization: don't call `on_send` unless the method name is in this list
+    RESTRICT_ON_SEND = %i[has_many belongs_to].freeze
+    ASSOCIATION_METHODS = RESTRICT_ON_SEND.to_set
+
+    def_node_matcher :association_expression_no_arguments?, <<~PATTERN
+      (send nil? ASSOCIATION_METHODS (sym _))
+    PATTERN
+
+    def_node_matcher :association_expression_with_arguments?, <<~PATTERN
+      (send nil? ASSOCIATION_METHODS (sym _) (hash $...))
+    PATTERN
+
+    def on_send(node)
+      return add_offense(node) if association_expression_no_arguments?(node)
+
+      return unless (hash_pairs = association_expression_with_arguments?(node))
+
+      add_offense(node) unless contain_inverse_of?(hash_pairs)
```

Comment:
It's more readable I guess, and using `if` reduces cognitive loads when reading the code 😂 

```suggestion
      hash_pairs = association_expression_with_arguments?(node)

      add_offense(node) if hash_pairs && contain_inverse_of?(hash_pairs)
```

Reply:
You're spot on. Fixed in fbdd5c688803fd1af1b92c3c8245029c07ae8c18